src/Symfony/Component/Form/Extension/Core/Type/ChoiceType.php

```diff
@@ -252,6 +253,39 @@ public function configureOptions(OptionsResolver $resolver)
             return $options['required'] ? null : '';
         };
 
+        // BC closure, to be removed in 3.0
+        $choicesNormalizer = function (Options $options, $choices) use (&$choiceLabels) {
+            // Unset labels from previous invocations
+            $choiceLabels = array();
+
+            // This closure is irrelevant when "choices_as_values" is set to true
+            if ($options['choices_as_values']) {
+                return $choices;
+            }
+
+            ChoiceType::normalizeLegacyChoices($choices, $choiceLabels);
+
+            return $choices;
+        };
+
+        // BC closure, to be removed in 3.0
+        $choiceLabel = function (Options $options) use (&$choiceLabels) {
```

Comment:
I know, this is not really beautiful. But the closures need to access the same variable to be able to properly exchange information.